### PR TITLE
Rename define to JACK_ON_WINDOWS

### PR DESCRIPTION
**Pull request by pljones**
**Short description of changes**
Renames a define.

CHANGELOG: Rename existing define to JACK_ON_WINDOWS

**Context: Fixes an issue?**
Renames a define.  Makes it consistent with the CONFIG option.

**Does this change need documentation? What needs to be documented and how?**
No.

**Status of this Pull Request**
Ready.

**What is missing until this pull request can be merged?**
Ready.

## Checklist

- [ ] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
- [ ] I tested my code and it does what I want
- [x] My code follows the [style guide](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#source-code-consistency) 
- [ ] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. 
- [x] I've filled all the content above

### CODE REVIEW

File: Jamulus.pro [hunk 1]
```diff
@@ -107,7 +107,7 @@ win32 {
         HEADERS += linux/sound.h
```
Comment (pljones) [inline Jamulus.pro:107]: Should line 104 stop the build?  It's not going to work, after all...

Comment (hoffie) [reply Jamulus.pro:107]: I'd say yes.

**Review by hoffie — APPROVED** [review]
Yep, that's certainly better than before.

In general, I was thinking whether linux/ should be renamed to jack/ (or at least parts of it? haven't checked). It's really strange to add linux/ on Windows or Mac builds.

I'm also wondering whether JACK_REPLACES_ASIO/JACK_ON_WINDOWS/JACK_ON_MAC could go away at some point (after all, there's WITH_JACK and there are defines for the actual platform as well, so I don't see the point of having one extra define for JACK on each platform).

All of that exceeds the scope of this PR though, so just adding this here as a comment.

cc @henkdegroot who last touched the JACK autobuild logic at least (although not directly related to the actual code parts).

File: Jamulus.pro (hunk 1 above)
Comment (pljones) [reply Jamulus.pro:107]: Mm, I'll do that in another PR.

**Review by ann0see — APPROVED** [review]

**Comment by ann0see** [comment]
Merged since it's a trivial change.

File: Jamulus.pro (hunk 1 above)
Comment (pljones) [reply Jamulus.pro:107]: If I remember...

**Pull request merged**